app/views/site/index.html.erb

```diff
@@ -29,7 +29,11 @@
         Ensure any updates to this ID are reflected in the ReactDOM.render call
         in app/javascript/packs/application.js
       #%>
-      <div id="home-page-content"></div>
+      <% if User.feature_flag_for_instance(current_user, 'cross_dataset_search') && @selected_branding_group.nil? %>
+        <div id="home-page-content-xds"></div>
+      <% else %>
+        <div id="home-page-content"></div>
```

Comment:
For the frontend flag, I'd prefer the feature flag branch to be controlled from JS, rather than rails.  This both lets us have less logic inside rails templates, and fewer 'magic strings' that have to be shared between rails and our JS.   So I'd create a new, trivial component that just handles reading the feature flag and rendering the appropriate search content.  (then you may want to rename the existing HomePageContent.js file to 'ClassicSearch.js' or somesuch, so that the new, trivial, component can just be called "HomePageContent"